src/handler/session.rs

```diff
@@ -130,7 +136,9 @@ impl Session {
     /// Generates session keys from an authentication header. If the IP of the ENR does not match the
     /// source IP address, we consider this session untrusted. The output returns a boolean which
     /// specifies if the Session is trusted or not.
+    #[allow(clippy::too_many_arguments)]
```

Comment:
While fixing the _clippy_ warnings/errors, this function failed because it has "too many arguments". Truly solving this may require deeper refactoring, so I opted to ignore the warning for this specific method.